sw/device/tests/sensor_ctrl_alerts.c

```diff
@@ -182,6 +169,26 @@ bool test_main(void) {
                                              /*ecc_en*/ false,
                                              /*he_en*/ false);
 
+  // If all bits are set, then this must be the first iteration
+  uint32_t max = (1 << SENSOR_CTRL_PARAM_NUM_ALERT_EVENTS) - 1;
+  uint32_t new_val;
+  if (events_vector == -1) {
```

Comment:
I think events_vector will always be -1 after each reset, so it seems you need to read the value from flash, or else you will be randomizing it on each reset, and will end up testing only the lowest bit set, so perhaps hardly testing the higher alerts.

Reply:
o no that's not the case. Because this is flash, it actually is only -1 on the "first" iteration. 
I've actually confirmed this on both fpga / sims.

Reply:
o i should also clarify. The events vector is directly mapped. Meaning you don't need the flash read routines to actually get the value. The processor can just directly read the address.